datalab/datalab_session/tests/test_operations.py

```diff
@@ -0,0 +1,240 @@
+    test_median = 'datalab/datalab_session/tests/test_files/median/median_1_2.fits'
+    test_fits_1 = 'datalab/datalab_session/tests/test_files/median/fits_1.fits.fz'
+    test_fits_2 = 'datalab/datalab_session/tests/test_files/median/fits_2.fits.fz'
+
+    def tearDown(self):
+        if os.path.exists(self.temp_median):
+            os.remove(self.temp_median)
+        return super().tearDown()
+
+    @mock.patch('datalab.datalab_session.file_utils.tempfile.NamedTemporaryFile')
+    @mock.patch('datalab.datalab_session.file_utils.get_fits')
+    @mock.patch('datalab.datalab_session.data_operations.median.save_fits_and_thumbnails')
+    @mock.patch('datalab.datalab_session.data_operations.median.create_jpgs')
+    def test_operate(self, mock_create_jpgs, mock_save_fits_and_thumbnails, mock_get_fits, mock_named_tempfile):
+
+        mock_get_fits.side_effect = [self.test_fits_1, self.test_fits_2]
+        mock_named_tempfile.return_value.name = self.temp_median
+        mock_create_jpgs.return_value = ('test_path', 'test_path')
+        mock_save_fits_and_thumbnails.return_value = self.temp_median
+
+        input_data = {
+            'input_files': [
+                {'basename': 'fits_1','source': 'local'},
+                {'basename': 'fits_2','source': 'local'}
+            ]
+        }
+
+        median = Median(input_data)
+        median.operate()
+        output = median.get_output().get('output_files')
+
+        self.assertEqual(median.get_percent_completion(), 1.0)
+        self.assertTrue(os.path.exists(output[0]))
+        self.assertEqual(md5(open(self.test_median, 'rb').read()).hexdigest(), md5(open(output[0], 'rb').read()).hexdigest())
+
+    def test_not_enough_files(self):
+        input_data = {
+            'input_files': [
+                {'basename': 'sample_lco_fits_1'}
+            ]
+        }
+        median = Median(input_data)
+        
+        with self.assertRaises(ClientAlertException):
+            median.operate()
+
+class TestRGBStackOperation(TestCase):
+    temp_rgb = 'datalab/datalab_session/tests/test_files/rgb_stack/temp_rgb.fits'
+    test_rgb = 'datalab/datalab_session/tests/test_files/rgb_stack/rgb_stack.fits'
+    test_red = 'datalab/datalab_session/tests/test_files/rgb_stack/red.fits'
+    test_green = 'datalab/datalab_session/tests/test_files/rgb_stack/green.fits'
+    test_blue = 'datalab/datalab_session/tests/test_files/rgb_stack/blue.fits'
+
+    def tearDown(self):
+        if os.path.exists(self.temp_rgb):
+            os.remove(self.temp_rgb)
+        return super().tearDown()
+    
+    @mock.patch('datalab.datalab_session.data_operations.rgb_stack.save_fits_and_thumbnails')
+    @mock.patch('datalab.datalab_session.data_operations.rgb_stack.create_jpgs')
+    @mock.patch('datalab.datalab_session.file_utils.tempfile.NamedTemporaryFile')
+    @mock.patch('datalab.datalab_session.data_operations.rgb_stack.get_fits')
```

Comment:
For the future: this many mocks maaaay indicate that we can refactor and streamline a few of these functions. Lots of mocks can indicate complexity - but this is a complex test.